### DISCUSSION TITLE
Forward directly to NPM registry

### DISCUSSION THREAD

**Discussion by christianhood** [verdaccio/verdaccio#3098] [Q&A] [answered]
Is there a way for verdaccio to tell the client to download official packages straight from the NPM registry URL when a package isn't found in the private registry? This would be rather than verdaccio downloading and serving that file. I would like to save on bandwidth on the server.

**Comment by juanpicado** [comment, marked as answer]
Nop, in any case that would be a client feature.

**Reply by edclement** [reply]
As Juan mentioned, you can accomplish this using an [.npmrc](https://docs.npmjs.com/cli/v8/configuring-npm/npmrc) file on the client. All of your private packages would need to be scoped. If your `.npmrc` file looked like:
```text
@your-scope:registry=https://registry.yourdomain.com/
//registry.yourdomain.com/:_authToken=abcd1234
//registry.yourdomain.com/:always-auth=true
```

...then the `npm` cli would install packages scoped to `@your-scope/` from registry.yourdomain.com and all other packages from the official public npm registry. You can add multiple scopes to the `.npmrc` file.

**Reply by christianhood** [reply]
Understood, unfortunately I can't do scoped packages as I'm not in charge of the name of the package here but thank you for confirming it isn't possible on the server side. I was hoping if the server returned the npmjs url the client would be fine with that.

**Reply by edclement** [reply]
Have you thought about putting an nginx in front of Verdaccio and setting up routing rules to forward only specific requests for packages to Verdaccio and then wildcard all other requests to the official npm repo?

An even potentially hackier solution would be to write your own [middleware plugin](https://verdaccio.org/docs/plugin-middleware/) that responds with a 301 redirect depending on the package name. I'm not sure it is actually viable. It would depend on:
1. The npm client respecting the 301 redirect response
2. Any logic/processing Verdaccio applies _before_ the middleware function responds (ensuring the express `next` function is not called).

The optimal solution would still be client side `.npmrc` configuration.

**Reply by christianhood** [reply]
I will look into those options and see what will work best for us. I appreciate the help with this, thank you!

**Reply by juanpicado** [reply]
@christianhood in adition what @edclement said, only works with scoped packages, which if is a private package is totally recommended, use scoped pkgs, so if you are relying on private non scoped packages that solution won´t work.
